src/CommunityToolkit/Xamarin.CommunityToolkit/Extensions/PageExtension.shared.cs

```diff
@@ -10,7 +10,7 @@ namespace Xamarin.CommunityToolkit.Extensions
 {
 	public static class PageExtension
 	{
-		public static Task DisplayToastAsync(this Page page, string message, int durationMilliseconds = 3000)
+		public static async Task DisplayToastAsync(this Page page, string message, int durationMilliseconds = 3000)
```

Comment:
There are very few differences between the two approaches. Basically, they share the same semantics. However, the version with async/await wraps the execution of the inner task in an outer compiler-generated task. The non-async version does not. Thus, the non-async version is (very marginally) more efficient.

Reply:
Yes, it's more efficient, but please, read the issue description. Since we currently don't use `await` in those methods exceptions from them are not wrapped inside a task and are thrown immediately (even task is not awaited).
This is the problem, because we don't always want to await tasks. For example, there are "fire and forget" methods that accept that task, and that await the task inside try-chatch (and can perform some action on exception for example).

Those methods don't work with current version of "DisplayXxx" methods since we throw exception immediately and never even call "fire and forget" methos.

See repro steps and demo code in issue description.